e2e_test/e2e_transfer_test.go

```diff
@@ -153,12 +173,12 @@ func TestTransferCELO(t *testing.T) {
 			blockNum, err := client.BlockNumber(ctx)
 			require.NoError(t, err)
 			signer := types.MakeSigner(devAccounts[0].ChainConfig, new(big.Int).SetUint64(blockNum))
-			tx, err := prepareTransaction(*tc.txArgs, sender.Key, sender.Address, signer, client)
+			tx, err := prepareTransaction(*tc.txArgs, sender.Key, sender.Address, signer, client, true)
 			require.NoError(t, err)
 			err = client.SendTransaction(ctx, tx)
 			if tc.expectedErr != nil {
 				// Once the error is checked, there's nothing more to do
-				if err.Error() != tc.expectedErr.Error() {
+				if err == nil || err.Error() != tc.expectedErr.Error() {
```

Comment:
Since we are inside the `if tc.expectedErr != nil` branch, the new check is redundant. Or is it somehow possible for
`tc.expectedErr != nil` and`tc.expectedErr.Error() == nil`?

Reply:
If for some reason, the tx is not throwing an error, you'll get a panic, instead of "Expected <something>, actual <nil>"
So yes, this is not strictly important, but for future modifications if we have an issue and those txs don't fail, I prefer that nil, and not a panic